packages/manager/src/features/IAM/Roles/RolesTable/AssignSelectedRolesDrawer.tsx

```diff
@@ -0,0 +1,178 @@
+import { useAccountUsers } from '@linode/queries';
+import { ActionsPanel, Autocomplete, Drawer, Typography } from '@linode/ui';
+import { useTheme } from '@mui/material';
+import Grid from '@mui/material/Grid2';
+import React, { useEffect, useState } from 'react';
+import { FormProvider, useFieldArray, useForm } from 'react-hook-form';
+
+import { LinkButton } from 'src/components/LinkButton';
+import { StyledLinkButtonBox } from 'src/components/SelectFirewallPanel/SelectFirewallPanel';
+import { AssignSingleSelectedRole } from 'src/features/IAM/Roles/RolesTable/AssignSingleSelectedRole';
+import {
+  useAccountPermissions,
+  useAccountUserPermissions,
+  useAccountUserPermissionsMutation,
+} from 'src/queries/iam/iam';
+
+import { mergeAssignedRolesIntoExistingRoles } from '../../Shared/utilities';
+
+import type {
+  AssignNewRoleFormValues,
+  RolesType,
+} from '../../Shared/utilities';
+import type { RoleView } from 'src/features/IAM/Shared/types';
+
+interface Props {
+  onClose: () => void;
+  open: boolean;
+  selectedRoles: RoleView[];
+}
+
+export const AssignSelectedRolesDrawer = ({
+  onClose,
+  open,
+  selectedRoles,
+}: Props) => {
+  const theme = useTheme();
+
+  const { data: allUsers } = useAccountUsers({});
+  const [username, setUsername] = useState<string>('');
+
+  const getUserOptions = () => {
+    return allUsers?.data.map((user) => ({
+      label: user.username,
+      value: user.username,
+    }));
+  };
+
+  const { data: accountPermissions } = useAccountPermissions();
+
+  const { data: existingRoles } = useAccountUserPermissions(username ?? '');
+
+  const form = useForm<AssignNewRoleFormValues>({
+    defaultValues: {
+      roles: [
+        {
+          entities: null,
+          role: null,
+        },
+      ],
+    },
+  });
+  const { control, handleSubmit, reset } = form;
+  const { update } = useFieldArray({
+    control,
+    name: 'roles',
+  });
+
+  // Add the roles that were selected
+  useEffect(() => {
+    selectedRoles.forEach((role, idx) => {
+      update(idx, {
+        entities: null,
+        role: role.name as unknown as RolesType,
+      });
+    });
+  }, [selectedRoles, update]);
```

Comment:
We should eliminate the useEffect by using react-hook-form's values option

```suggestion

  const values = {
    roles: selectedRoles.map((r) => ({ role: r.name, entities: null })),
  };

  const form = useForm<AssignNewRoleFormValues>({
    defaultValues: values,
    values,
  });
```

Reply:
Thanks, I accepted this suggestion.  Asking as a newbie - is `useEffect` undesirable?